data_processors/pipeline/services/workflow_srv.py

```diff
@@ -59,38 +55,66 @@ def create_or_update_workflow(model: dict):
         else:
             workflow.start = start  # expect start in raw zone-aware UTC datetime string e.g. "2020-06-25T10:45:20.438Z"
 
-        if model.get('end_status'):
-            workflow.end_status = model.get('end_status')
-
     else:
 
-        logger.info(f"Updating existing {wfl_type.name} workflow (wfl_id={wfl_id}, wfr_id={wfr_id}, wfv_id={wfv_id})")
+        logger.info(f"Updating existing {wfl_type.value} workflow (portal_run_id={portal_run_id})")
         workflow: Workflow = qs.get()
 
+        # --- business logic: the following fields are only settable at "update" time
+        # only mutable if the specified record has already existed
+
+        workflow.notified = model.get('notified')
+
         _output = model.get('output')
         if _output:
             if isinstance(_output, dict):
                 workflow.output = libjson.dumps(_output)  # if output is in dict
             else:
                 workflow.output = _output  # expect output in raw json str
 
-        if model.get('end_status'):
-            workflow.end_status = model.get('end_status')
-
         _end = model.get('end')
         if _end:
             if isinstance(_end, datetime):
                 workflow.end = _end if is_aware(_end) else make_aware(_end)
             else:
                 workflow.end = _end  # expect end in raw zone-aware UTC datetime string e.g. "2020-06-25T10:45:20.438Z"
 
-        workflow.notified = model.get('notified')
+    # --- business logic: below are mutable fields regardless of new or existing records, if they have been payload
```

Comment:
Mutable yes, but ideally only set once then should match anything that follows, right?
(could be future improvement)

Reply:
Okay! Noted well. Let us observe and improve it in next iteration.